twill-yarn/src/main/java/org/apache/twill/internal/appmaster/RunningContainers.java

```diff
@@ -113,9 +117,11 @@ public Integer apply(BitSet input) {
   private final Location applicationLocation;
   private final Set<String> runnableNames;
   private final Map<String, Map<String, String>> logLevels;
+  private final Map<String, Integer> maxRetries;
```

Comment:
@poornachandra, I don't see anything linking one instance request to a retry of that failure.  Failures get added to the multiset which then results in a new request for that many instances.  This means in the end, there is nothing tying one attempt for a particular instanceId to the next one.  So we could keep track of retries by instanceId, but it would end up functionally being the same as keeping track of retries for the entire Runnable.  I think it would just add more complexity for little benefit to track by instanceId.